modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java

```diff
@@ -120,106 +128,123 @@ public BulkResponseData processBulkRequests(BulkRequestData bulkRequestData) thr
+                    bulkResponseData.addRoleOperation(getBulkResponseContent(bulkRequestContent,
+                            userIdMappings, roleResourceManager));
                 }
             }
         }
         bulkResponseData.setSchema(SCIMConstants.BULK_RESPONSE_URI);
         return bulkResponseData;
     }
 
+    private BulkResponseContent getBulkResponseContent(BulkRequestContent bulkRequestContent,
+                                                       ResourceManager resourceManager) throws BadRequestException {
 
-   private BulkResponseContent getBulkResponseContent
-           (BulkRequestContent bulkRequestContent, ResourceManager resourceManager)
-           throws BadRequestException {
-
-       BulkResponseContent bulkResponseContent = null;
-       SCIMResponse response;
-
-       if (bulkRequestContent.getMethod().equals(SCIMConstants.OperationalConstants.POST)) {
-
-           if (bulkRequestContent.getPath().contains(SCIMConstants.ROLE_ENDPOINT)) {
-               response = resourceManager.createRole(bulkRequestContent.getData(), roleManager);
-           } else {
-               response = resourceManager.create(bulkRequestContent.getData(), userManager, null, null);
-           }
-
-           bulkResponseContent = createBulkResponseContent
-                   (response, SCIMConstants.OperationalConstants.POST, bulkRequestContent);
-           errorsCheck(response);
-
-       } else if (bulkRequestContent.getMethod().equals(SCIMConstants.OperationalConstants.PUT)) {
-
-           String resourceId = extractIDFromPath(bulkRequestContent.getPath());
-           if (bulkRequestContent.getPath().contains(SCIMConstants.ROLE_ENDPOINT)) {
-               response = resourceManager.updateWithPUTRole(resourceId, bulkRequestContent.getData(), roleManager);
-           } else {
-               response = resourceManager
-                       .updateWithPUT(resourceId, bulkRequestContent.getData(), userManager, null, null);
-           }
+        return getBulkResponseContent(bulkRequestContent, null, resourceManager);
+    }
 
-           bulkResponseContent = createBulkResponseContent
-                   (response, SCIMConstants.OperationalConstants.PUT, bulkRequestContent);
-           errorsCheck(response);
+    private BulkResponseContent getBulkResponseContent(BulkRequestContent bulkRequestContent,
+                                                       Map<String, String> userIdMappings,
+                                                       ResourceManager resourceManager)
+            throws BadRequestException {
+
+        BulkResponseContent bulkResponseContent = null;
+        SCIMResponse response;
+        processBulkRequestContent(bulkRequestContent, userIdMappings, bulkRequestContent.getMethod());
+
+        switch (bulkRequestContent.getMethod()) {
+            case SCIMConstants.OperationalConstants.POST:
+                if (bulkRequestContent.getPath().contains(SCIMConstants.ROLE_ENDPOINT)) {
```

Comment:
bulkRequestContent.getPath() could this be null?

Reply:
No. Before coming to the `BulkRequestProcessor` these `bulkRequestContent` objects are created in `JSONDecoder`. The `requestType` is set as the path of `bulkRequestContent` [here](https://github.com/PasinduYeshan/wso2-charon/blob/92a75e1d2e095af77d13fac15c28ff3ab112dea8/modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONDecoder.java#L1007).  Therefore, path can only be null if `requestType` is null, but this check is done [here](https://github.com/PasinduYeshan/wso2-charon/blob/92a75e1d2e095af77d13fac15c28ff3ab112dea8/modules/charon-core/src/main/java/org/wso2/charon3/core/encoder/JSONDecoder.java#L910-L913) and if the `requestType` is null an error is thrown.